src/go/util/marshal_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+
+	confpb "google.golang.org/genproto/googleapis/api/serviceconfig"
+	scpb "google.golang.org/genproto/googleapis/api/servicecontrol/v1"
+	smpb "google.golang.org/genproto/googleapis/api/servicemanagement/v1"
+)
+
+func TestUnmarshalBytesToPbMessage(t *testing.T) {
+	var wantResp, getResp proto.Message
+	wantError := ""
+
+	_test := func() {
+		bytes, _ := proto.Marshal(wantResp)
+		err := UnmarshalBytesToPbMessage(bytes, getResp)
+		if err != nil {
+			if err.Error() != wantError {
+				t.Fatalf("fail in UnmarshalBytesToPbMessage, want error: %s, get error: %v", wantError, err)
+			}
+			return
+		}
+		if !proto.Equal(getResp, wantResp) {
+			t.Errorf("fail in UnmarshalBytesToPbMessage, want: %v, ge: %v", wantResp, getResp)
+		}
+	}
+
+	wantResp = &confpb.Service{
+		Id: "test-id",
+	}
+	getResp = &confpb.Service{}
+	_test()
```

Comment:
can we separate them into different tests?

Reply:
What do you mean by separating them into different tests? 
I feel those test cases belong to same category.

Reply:
Into different Test*** so if one of them fails, the error message is clear.  For now, if one of them fail, the whole test fails.  have to read the log to find out which test fails.

Reply:
- Replace Fatal with Error so one failure won't effect others. 
- Print the type of marshal target, so it looks like following and it is straightforward.
`marshal_test.go:36: fail in UnmarshalBytesToPbMessage on *servicecontrol.ReportRequest...`

Reply:
> The logic for config_manager and RolloutDetector and ConfigFetcher are good.
> 
> Is the test code updated for the new logic?

Done.